Add another test and upgrade guide for #36285 (#36635)
gradle/gradle: platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_9.adoc

. Run `gradle wrapper --gradle-version {gradleVersion}` to update the project to {gradleVersion}.
. Try to run the project and debug any errors using the <<troubleshooting.adoc#troubleshooting, Troubleshooting Guide>>.

[[changes_9.5.0]]
== Upgrading from 9.4.0 and earlier

=== Potential breaking changes

==== Plugins requested by precompiled settings plugins are now resolved in the build that compiles them

> If a precompiled settings plugin (`*.settings.gradle.kts`) requests a plugin in the `plugins { }` block, the

Reviewer: ```suggestion
When a precompiled settings plugin (`*.settings.gradle.kts`) declares a plugin in a `plugins {}` block, that
```

plugin request is now going to be checked against the dependencies in the build that compiles the plugin.

> Previously, if a precompiled settings plugin was never applied in the build, it could contain an invalid

Reviewer: ```suggestion
Previously, if a precompiled settings plugin was never applied, it could contain an invalid plugin request
```

[[deprecate_apply_false_in_precompiled_script_plugins]]
==== Deprecation of `apply false` in precompiled script plugins
The use of `apply false` in <<implementing_gradle_plugins_precompiled.adoc#implementing_precompiled_plugins, precompiled script plugins>> has been deprecated and will become an error in Gradle 10.0.0.

Using `apply false` does not have any effect in precompiled script plugins (the plugin is still applied), and it can lead to confusion for users who expect it to prevent the plugin from being applied.
Plugins used in precompiled script plugins are already available on the classpath, so using `apply false` would do nothing.

To fix this deprecation, remove the plugin declaration from the precompiled script plugin:
.my-plugin.gradle.kts
[source,kotlin]
----
plugins {
    // Delete the following line to remove the deprecation warning:
    id("org.gradle.test-retry") apply false
    // The plugin will still be on the classpath, but it will not be applied as part of this precompiled script plugin.
}
----

[[deprecate_version_in_precompiled_settings_script_plugins]]

> ==== Deprecation of `version` in precompiled settings script plugins

Reviewer: Fixed in https://github.com/gradle/gradle/pull/36669
